docs/outputs.md

```diff
@@ -59,3 +59,108 @@ which will produce a ```png``` image per dump suitable for encoding into a movie
+optional and are only used for output to HDF5. 4D variables are named with a
+list of names for each row while 3D variables are named with a single name.
+For example, the following configurations are acceptable:
+
+```c++
+auto pkg = std::make_shared<StateDescriptor>("Hydro");
+
+/* ... */
+const int nhydro = 5;
+std::vector<std::string> cons_labels(nhydro);
+cons_labels[0]="Density";
+cons_labels[1]="MomentumDensity1";
+cons_labels[2]="MomentumDensity2";
+cons_labels[3]="MomentumDensity3";
+cons_labels[4]="TotalEnergyDensity";
+Metadata m({Metadata::Cell, Metadata::Independent, Metadata::FillGhost},
+           std::vector<int>({nhydro}), cons_labels);
+pkg->AddField("cons", m);
+
+const int ndensity = 1;
+std::vector<std::string> density_labels(ndensity);
+density_labels[0]="Density";
+m = Metadata({Metadata::Cell, Metadata::Derived}, std::vector<int>({ndensity}), density_labels);
+pkg->AddField("dens", m);
+
+const int nvelocity = 3;
+std::vector<std::string> velocity_labels(nvelocity);
+velocity_labels[0]="Velocity1";
+velocity_labels[1]="Velocity2";
+velocity_labels[2]="Velocity3";
+m = Metadata({Metadata::Cell, Metadata::Derived}, std::vector<int>({nvelocity}), velocity_labels);
+pkg->AddField("vel", m);
+
+const in npressure = 1;
+std::vector<std::string> pressure_labels(npressure);
+pressure_labels[0]="Pressure";
+m = Metadata({Metadata::Cell, Metadata::Derived}, std::vector<int>({npressure}), pressure_labels);
+pkg->AddField("pres", m);
+```
+
+The `yt` frontend needs either the hydrodynamic conserved variables or
+primitive compute derived quantities. The conserved variables must have the
+names ```"Density"```, ```"MomentumDensity1"```, ```"MomentumDensity2"```,
+```"MomentumDensity3"```, ```"TotalEnergyDensity"``` while the primitive
+variables must have the names ```"Density"```, ```"Velocity1"```,
+```"Velocity2"```, ```"Velocity3"```, ```"Pressure"```. Either of these sets of
+variables must be named and present in the output, with the primitive variables
+taking precedence over the conserved variables when computing derived
+quantities such as specific thermal energy. In the above example, including
+either ```"cons"``` or ```"dens"```, ```"vel"```, and ```"pres"``` in the  HDF5
+output would allow ```yt``` to read the data.
+
+Additional parameters can also be packaged into the HDF5 file to help ```yt```
+interpret the data, namely adiabatic index and code unit information. These are
+identified by passing ```true``` as an optional boolean argument when adding
+parameters via ```StateDescriptor::AddParam```. For example, 
+```c++
+pkg->AddParam<double>("CodeLength", 100,true);
+pkg->AddParam<double>("CodeMass", 1000,true);
+pkg->AddParam<double>("CodeTime", 1,true);
+pkg->AddParam<double>("AdibaticIndex", 5./3.,true);
+
+pkg->AddParam<int>("IntParam", 0,true);
+pkg->AddParam<std::string>("EquationOfState", "Adiabatic",true);
+```
+all function to add parameters to the HDF5 output. Currently, only ```int```,
```

Comment:
I'll change to "add the parameters `CodeLength`, `CodeMass`, `CodeTime`, `AdiabaticIndex`, `IntParam`, and `EquationOfState` to the HDF5 output" since it's more descriptive.